lib/orm.php

```diff
@@ -2196,6 +2196,82 @@ public function save() {
 		return $success;
 	}
 
+	/**
+	 * Inserts multiple rows in a single query. Expects new rows as it's a strictly insert function, not an update one.
+	 *
+	 * @param array $models Array of model instances to be inserted.
+	 *
+	 * @return bool True for successful insert, false for failed.
+	 *
+	 * @example From the Indexable_Link_Builder class: $this->seo_links_repository->query()->insert_many( $links );
+	 *
+	 * @throws \InvalidArgumentException Invalid instances to be inserted.
+	 * @throws \InvalidArgumentException Instance to be inserted is not a new one.
+	 */
+	public function insert_many( $models ) {
+		// Validate the input first.
+		if ( ! \is_array( $models )  ) {
+			throw new \InvalidArgumentException( 'Invalid instances to be inserted' );
+		}
+
+		if ( empty( $models ) ) {
+			return false;
+		}
+
+		$total_dirty_fields = [];
+		$success            = true;
+
+		// First, we'll gather all the dirty fields throughout the models to be inserted.
+		foreach ( $models as $model ) {
+			if ( ! $model->orm->is_new ) {
+				throw new \InvalidArgumentException( 'Instance to be inserted is not a new one' );
+			}
+
+			// Remove any expression fields as they are already baked into the query.
+			$dirty_fields       = \array_diff_key( $model->orm->dirty_fields, $model->orm->expr_fields );
+			$total_dirty_fields = \array_merge( $total_dirty_fields, $dirty_fields );
+		}
+
+		$total_dirty_fields = \array_keys( $total_dirty_fields );
+
+		/**
+		 * Filter: 'wpseo_chunk_bulked_insert_queries' - Allow filtering the chunk size of each bulked INSERT query.
+		 *
+		 * @api int The chunk size of the bulked INSERT queries.
+		 */
+		$chunk = \apply_filters( 'wpseo_chunk_bulked_insert_queries', 100 );
+		$chunk = ! \is_int( $chunk ) ? 100 : $chunk;
+		$chunk = ( $chunk <= 0 ) ? 100 : $chunk;
+
+		$model_count = ( \count( $models ) );
+		while ( $model_count > 0 ) {
```

Comment:
I thought we discussed using `array_chunk` see [here](https://www.php.net/manual/en/function.array-chunk.php) 
that would make this a simpler ```foreach ( $chunks as $chunk ) {
 // do the chunk
}```
with less counting / slicing / tracking to do.
IMHO manipulating an array while you're looping over it is a high risk operation.